### ISSUE TITLE
Confusion when silently chdir to the directory where an `.env` file was found (with `python-dotenv`)

### ISSUE THREAD

**Issue by cjauvin** [pallets/flask#3535]
TL;DR: I think it would be helpful to add a `click.secho("Warning the current directory has changed to..")` there: https://github.com/pallets/flask/blob/master/src/flask/cli.py#L656

--

I have spent quite some time trying to understand what was going on in the following situation: suppose I have a project with multiple apps, among which a `toto` Flask app (with `python-dotenv` present in its virtual environment):

```
/project/apps/toto/__init__.py  <-- the `app` object lives there 
```

Because `toto` is part of a more general project, I want to centralize its configuration settings at the root of the project, in order for them to be shared among the different other apps and scripts (not sure that is such a good idea but that is what I've been experimenting with):

```
/project/.env  <-- configuration for `toto` lives there, among other things
```

Now the problem I have been having in this particular setup is that I did what I normally do while developing and testing a Flask app:

```
$ cd /project/apps/
$ FLASK_APP=toto flask run
```

and here the fun begins because Flask will silently `os.chdir` to the directory where the `.env` file was found, and in this particular case won't be able to load the app afterward because it no longer finds itself in the directory where `flask run` was ran, and thus won't be able to find `toto` there.I should add the detail that is happening even though `FLASK_APP` is *not* defined in `/project/.env`: the only place where I want it defined in this development context is on the command-line, at the moment I run it.

Now that I properly understand it, I find this behavior logical and helpful, and I am certainly not proposing to change it. However what would have clearly helped in my particular case was if `flask run` would have informed me about it (a simple console warning), and that is what I'd like to discuss with this issue, if anyone finds that it makes sense.

**Comment by davidism** [comment]
I really don't want to add more output to the dev server startup. The more we add, the less people will pay attention to things like the "don't use this in production" warning.

This behavior is documented in the CLI docs.

**Comment by davidism** [comment]
This is also an argument for using editable installs so importing your project doesn't depend on what the current directory is. The tutorial in the docs discusses how to do this.

See #3108 for a longer discussion on loading env files. I think if we do anything about this issue, it will be part of that issue, so I'm going to close this and leave a comment there.

**Comment by alexandersvozil** [comment]
Finally, after 2 hours I found my problem. This is absolutely ridiculous! Thank you for documenting this @cjauvin.
